### PR TITLE
Add geojson import support

### PR DESCRIPTION
**Pull request by JamesHovet**
Something I want your input on: right now the importer looks at the first feature within a featureCollection (exactly what it sounds like) and takes inventory of all the data properties within it and what data types they are and uses that to create the schema for the resulting VectorDataset. If any other feature has a different set of properties that doesn't match that schema, it throws an error. This system most matches up with how traditional shapefiles work, and any geojson file that was, say exported by QGIS or ArcMap isn't going to throw any errors because of this. 

However, I can think of a ways to approach this that would allow more files to be parsed without throwing errors, but could create situations where the user would rather have just been made aware that there might be an issue with their data and want to hear your take:

Go through every feature and take the union of all the sets of properties (throwing an error if there are two properties with the same name but different datatypes), create the schema like that, and only then go through every feature and parse their property values. If there are any places where one feature is missing a certain property, replace it with a default value. 

If we went in that direction, we could have cases where someone edits a geojson file by hand (which is certainly possible), accidentally mispells a key name and then our importer doesn't do anything to help them spot that mistake and they end up with a new "column" of data with one non-defaut row and an original column with a default value where there should be a real one. Additionally, I'm just always wary of putting a default value somewhere where you really would prefer to have the user step in and actually fix the problem

### CODE REVIEW

**Comment by LaCuneta** [comment]
Do you know how common it is to have features in a feature collection whose properties would be different?  If it's a really common occurrence I can your solution being a good alternative.  If it's relatively uncommon, I'm inclined to keep throwing an error when a property is missing/extra after the first feature, and maybe add a prim or argument that would let the "relaxed" loading occur?  One other possibility that is less common in NetLogo extensions would be to do the "relaxed" loading, but throw warnings to the command console whenever the property mismatches are found.  At least then model authors would get a chance to handle unexpected differences rather than wasting time debugging unexpected model results.

I hope that's helpful, we can discuss further next week, too.

File: README.md [hunk 1]
```diff
@@ -523,19 +523,19 @@ gis:load-dataset *file*
...
-assumes that the projection of the data being loaded is the same as
+For ESRI shapefiles and ESRI grid files, if there is a ".prj" file 
+associated with the file, then `load-datset` will consult that file
+and re-project to the current global projection if needed. If no ".prj"
+file is found, then the data is assumed to use the same projection as
 the current global coordinate system.
 
-Relative paths are resolved relative to the location of the current
-model, or the user's home directory if the current model
-hasn't been saved yet.
```
Comment (LaCuneta) [inline README.md:536]: I see that this relative path info is also listed in the `set-coordinate-system` docs.  Is there a reason you don't think it should be included here as well?  It's still valid even with the GeoJSON changes, correct?

File: src/org/myworldgis/io/geojson/GeoJsonReader.java [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    
+
+    public GeoJsonReader(File file, GeometryFactory factory) throws IOException, ParseException, ExtensionException {
+        this.factory = factory;
+        this.containsDefaultValues = false;
+        this.propertyNamesToDatatypes = new HashMap<String, PropertyType>();
+
+        // //TODO: Figure out why file.reader() wouldn't go into parser.parse. 
+        // Scanner s = new Scanner(file.getInputStream()).useDelimiter("\\Z");
+        // String out = s.hasNext() ? s.next() : "";
```
Comment (LaCuneta) [inline src/org/myworldgis/io/geojson/GeoJsonReader.java:64]: Is this TODO still relevant?

**Review by LaCuneta — COMMENTED** [review]
Nice work!  I tried this out with [a random GeoJSON dataset I found online](https://www.nps.gov/lib/npmap.js/4.0.0/examples/data/national-parks.geojson) and it worked fine.

I like seeing the different test files there, too, and they served as good examples for me to use with my random data.
 
Is it worth thinking about adding GeoJSON as an option for `store-dataset`?  Maybe if `json` or `geojson` is the file extension?  I'm not sure how much that prim gets, honestly, and I don't think we'd have to hold up merging this PR for it, just wanted to get your thoughts.

I had one small question on the docs, left in code, otherwise this looks good to me and we can merge once that is addressed.

**Comment by LaCuneta** [comment]
And I forgot one more thing in my review, it looks like this branch may need to be rebased against `hexy` as we did before - there are conflicts in the `tests.txt` file again.

**Comment by JamesHovet** [comment]
> thinking about adding GeoJSON as an option for `store-dataset`? Maybe if `json` or `geojson` is the file extension? I'm not sure how much that prim gets, honestly, and I don't think we'd have to hold up merging this PR for it, just wanted to get your thoughts.

I am totally in favor of adding it -- would feel lopsided if we didn't and I also agree it can be its own PR

**Comment by JamesHovet** [comment]
Merged, but not through this PR.

**Pull request closed without merging**